spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClient.java

```diff
@@ -14,132 +14,161 @@
+        ServiceInstance defaultInstance = new DefaultServiceInstance(serviceName,
+                "localhost",
+                8080,
+                false);
+
+        Endpoints endpoints = client.endpoints().withName(serviceName).get();
+        Optional<Service> service = Optional.ofNullable(client.services().withName(serviceName).get());
+        final Map<String, String> labels;
+        if (service.isPresent()) {
+            labels = service.get().getMetadata().getLabels();
+        } else {
+            labels = null;
+        }
+        if (Utils.isNullOrEmpty(podName) || endpoints == null) {
+            return defaultInstance;
+        }
+        try {
+            List<EndpointSubset> subsets = endpoints.getSubsets();
+
+            if (subsets != null) {
+                for (EndpointSubset s : subsets) {
+                    List<EndpointAddress> addresses = s.getAddresses();
+                    for (EndpointAddress a : addresses) {
+                        return new KubernetesServiceInstance(serviceName,
+                                a,
+                                s.getPorts().stream().findFirst().orElseThrow(IllegalStateException::new),
+                                labels,
+                                false);
+                    }
+                }
+            }
+            return defaultInstance;
+
+        } catch (Throwable t) {
+            return defaultInstance;
+        }
+    }
+
+    @Override
+    public List<ServiceInstance> getInstances(String serviceId) {
+        Assert.notNull(serviceId,
+                "[Assertion failed] - the object argument must be null");
+        Optional<Service> service = Optional.ofNullable(client.services().withName(serviceId).get());
+        final Map<String, String> labels;
+        if (service.isPresent()) {
+            labels = service.get().getMetadata().getLabels();
+        } else {
+            labels = null;
+        }
+
+        Optional<Endpoints> endpoints = Optional.ofNullable(client.endpoints().withName(serviceId).get());
+        List<EndpointSubset> subsets = endpoints.get().getSubsets();
+        List<ServiceInstance> instances = new ArrayList<>();
+        if (subsets != null) {
+            for (EndpointSubset s : subsets) {
+                List<EndpointAddress> addresses = s.getAddresses();
+                for (EndpointAddress a : addresses) {
+                    instances.add(new KubernetesServiceInstance(serviceId,
+                            a,
+                            s.getPorts().stream().findFirst().orElseThrow(IllegalStateException::new),
+                            labels,
+                            false));
+                }
+            }
+        }
+
+        return instances;
+    }
+
+    @Override
+    public List<String> getServices() {
+        String spelExpresion = properties.getFilter();
+        Predicate<Service> filteredServices;
+        if (spelExpresion == null || spelExpresion.isEmpty()) {
```

Comment:
typo: spelExpression

Reply:
@paolocarta typo fixed. Not sure about StringUtils, is that used all over spring? I wouldn't want to introduce something that is not widely used in the framework.